system/doc/reference_manual/data_types.xml

```diff
@@ -87,6 +87,37 @@
 11> <input>1_234.333_333</input>
 1234.333333
 </pre>
+<marker id="numeric_comparisons" />
+    <section>
+      <title>Comparisons</title>
+      <p>Both integers and floats share the same linear order. That is,
+        <c>1</c> compares less than <c>2.4</c>, <c>3</c> compares greater than
+        <c>2.99999</c>, and <c>5</c> is equal to <c>5.0</c>.</p>
+      <p>When wanting to match a specific number without also matching its
+        integer- or floating-point twin, it may be tempting to use the term
+        equivalence operators (<c>=:=</c>, <c>=/=</c>) or pattern matching.
+        This works for integers which has a distinct representation for every
+        number, but does not work for floating-point as the latter has two
+        representations for zero which are considered different by the term
+        equivalence operator (<c>=:=</c>).</p>
+      <p>If you wish to specifically compare floating-point numbers, use the
+        regular comparison operators (such as <c>==</c>) and add guards that
+        require both the arguments to be floating-point.</p>
```

Comment:
Nitpick: I am not sure if we should label it as "does not work". As you said later, you are comparing them as terms, not as numbers, which is fine by IEEE754. To me this is 

```suggestion
        However, beware floating-point numbers have two representations for
        zero (<c>0.0</c> and <c>-0.0</c>) and the term equivalent operator
        (<c>=:=</c>) will consider them to be distinct, which may be undesired.
        This is not a concern for integers, which have a distinct representation
        for every number.</p>
      <p>If you wish to specifically compare floating-point numbers while
        keeping the zero equivalence, use the regular comparison operators
        (such as <c>==</c>) and add guards that require both the arguments
        to be floating-point.</p>
```

Reply:
Thanks, I've reworded it a bit :)